pilot/pkg/xds/sds.go

```diff
@@ -190,6 +130,15 @@ func (s *SecretGen) Generate(proxy *model.Proxy, push *model.PushContext, w *mod
 			}
 		}
 
+		// Fetch the appropriate cluster's secrets, based on the credential type
+		var secretController secrets.Controller
+		switch sr.Type {
+		case credentials.KubernetesGatewaySecretType:
```

Comment:
Should/would this API ever allow for cross-cluster references? Where the proxy is co-located with it's secrets.

Reply:
Good question. IMO no, it should always be the config cluster. A secret is just another config in this model.
I think this is how MCI works, which is the only real prior art I know of. Verifying with them now.

WDYT?